reyna/src/com/b2msolutions/reyna/Preferences.java

```diff
@@ -58,4 +65,78 @@ public void resetStorageSize() {
         edit.remove(STORAGE_SIZE);
         edit.apply();
     }
+
+    public void saveWlanBlackout(String value) {
+        if (isBlackoutRangeValid(value)) {
+            SharedPreferences sp = this.context.getSharedPreferences(Preferences.class.getName(), Context.MODE_PRIVATE);
+            SharedPreferences.Editor edit = sp.edit();
+            edit.putString(WLAN_RANGE, value);
+            edit.apply();
+        }
+    }
+
+    public String getWlanBlackout() {
+        SharedPreferences sp = this.context.getSharedPreferences(Preferences.class.getName(), Context.MODE_PRIVATE);
+        return sp.getString(WLAN_RANGE, "");
+    }
+
+    public void saveWwanBlackout(String value) {
+        if(isBlackoutRangeValid(value)) {
+            SharedPreferences sp = this.context.getSharedPreferences(Preferences.class.getName(), Context.MODE_PRIVATE);
+            SharedPreferences.Editor edit = sp.edit();
+            edit.putString(WWAN_RANGE, value);
+            edit.apply();
+        }
+    }
+
+    public String getWwanBlackout() {
+        SharedPreferences sp = this.context.getSharedPreferences(Preferences.class.getName(), Context.MODE_PRIVATE);
+        return sp.getString(WWAN_RANGE, "");
+    }
+
+    public void saveWwanRoamingBlackout(boolean value) {
+        SharedPreferences sp = this.context.getSharedPreferences(Preferences.class.getName(), Context.MODE_PRIVATE);
```

Comment:
We are validating data on save, so we will save data only when it is valid.

@lazywithclass